### PR TITLE
Prevent submitting a span to Exporters twice.

### PR DESCRIPTION
**Pull request by ramonza**
Fixes: #328

### CODE REVIEW

File: trace/trace_test.go [hunk 1]
```diff
@@ -582,3 +582,63 @@ func TestBucket(t *testing.T) {
...
+	UnregisterExporter(&spans)
+	if len(spans) != 3 {
+		t.Fatalf("expected 3 spans, got %#v", spans)
+	}
+	var span1, span2, parent *SpanData
+	for _, span := range spans {
+		switch span.Name {
+		case "span-1":
+			span1 = span
+			break
```
Comment (rakyll) [inline trace/trace_test.go]: You don't have to break after case body. Go default behavior is to break.

File: trace/trace_test.go [hunk 2]
```diff
@@ -582,3 +582,63 @@ func TestBucket(t *testing.T) {
...
+		case "span-2":
+			span2 = span
+			break
```
Comment (rakyll) [inline trace/trace_test.go]: ditto

File: trace/trace_test.go [hunk 3]
```diff
@@ -582,3 +582,63 @@ func TestBucket(t *testing.T) {
...
+		case "parent":
+			parent = span
+			break
```
Comment (rakyll) [inline trace/trace_test.go]: ditto

File: trace/trace_test.go [hunk 4]
```diff
@@ -582,3 +582,63 @@ func TestBucket(t *testing.T) {
...
+		}
+	}
+	if span1.TraceID != parent.TraceID || span2.TraceID != parent.TraceID {
+		t.Errorf("expected same TraceID, got: %q, %q, %q", parent.TraceID, span1.TraceID, span2.TraceID)
+	}
+	if span1.ParentSpanID != parent.SpanID {
+		t.Errorf("span-1 parent, expected: %q got: %q", parent.SpanID, span1.ParentSpanID)
```
Comment (rakyll) [inline trace/trace_test.go]: Use got, want terminology. Got first, want second.

You can omit got if you already have an identifier for what's got in the message:

"span-1 parent = %v; want %v"

File: trace/trace.go [hunk 5]
```diff
@@ -41,6 +41,7 @@ type Span struct {
 	spanContext SpanContext
 	// spanStore is the spanStore this span belongs to, if any, otherwise it is nil.
 	*spanStore
+	once sync.Once
```
Comment (odeke-em) [inline trace/trace.go]: Could we make this field more desscriptive like `exportOnce`?

File: trace/trace_test.go [hunk 6]
```diff
@@ -582,3 +582,63 @@ func TestBucket(t *testing.T) {
...
+	ctx1 := StartSpanWithOptions(ctx, "span-1", StartSpanOptions{Sampler: AlwaysSample()})
+	EndSpan(ctx1)
+	// Start a new span with the context containing span-1
+	// even though span-1 is ended, we still add this as a new child of span-1
+	ctx2 := StartSpanWithOptions(ctx1, "span-2", StartSpanOptions{Sampler: AlwaysSample()})
+	EndSpan(ctx2)
+	EndSpan(ctx)
+	UnregisterExporter(&spans)
+	if len(spans) != 3 {
+		t.Fatalf("expected 3 spans, got %#v", spans)
```
Comment (odeke-em) [inline trace/trace_test.go]: Ditto about what @rakyll said about the Go style of got and want.

**Pull request merged**